src/util/hass-call-api.js

```diff
@@ -1,28 +1,30 @@
-export default function hassCallApi(host, auth, method, path, parameters) {
-  var url = host + '/api/' + path;
+export default function hassCallApi(host, auth, method, path, parameters, responseType = '') {
+  const url = `${host}/api/${path}`;
 
   return new Promise(function (resolve, reject) {
     var req = new XMLHttpRequest();
+    req.responseType = responseType;
     req.open(method, url, true);
     req.setRequestHeader('authorization', `Bearer ${auth.accessToken}`);
 
     req.onload = function () {
-      let body = req.responseText;
-      const contentType = req.getResponseHeader('content-type');
+      var body = req.response;
+      if (responseType === '' || responseType === 'text') {
```

Comment:
Probably better to just do `if (!responseType)`. If `responseType` is explicitly passed as `text` we shouldn't try to parse json